### ISSUE TITLE
Promise.all with flush in each promise

### ISSUE THREAD

**Issue by notmedia** [mikro-orm/mikro-orm#2070]
**Describe the bug**
Hi, I don't understand why this is not working:

Having an `EntityService` with `create` and `handleError` methods:
```ts
  async create(payload: CreateEntity): Promise<Entity> {
    const entity = new Entity(...payload);

    await this.repository.persistAndFlush(entity);

    return entity;
  }

  async handleError(
    uuid: string,
    result: Result
  ): Promise<void> {
     // Error here because entity was not created
    const entity = await this.repository.findOneOrFail(uuid);

    entity.updateStatus(result);

    await this.repository.flush();
  }
```

Having another composite service `WorkerService` which uses `EntityService` inside
```ts
  async someWork(
    entityData: CreateEntity,
    payload: Record<string, unknown>,
  ): Promise<void> {
    const entity = await this.entityService.create(entityData);

    try {
      await this.anotherService.someStuff(entity, ...payload);

      return entity;
    } catch (error) {
     // Error here because entity was not created
      await this.entityService.handleError(entity.uuid, Result.FAILED);

      throw error;
    }
  }
```

And the last one service which handles multiple creations
```ts
  create(data: Data) {
    const promises = data.map((item) =>
      this.workerService.someWork(item.entityData, item.payload)
    );

    return Promise.all(promises);
}
```

stack:

```
"type": "Error",
      "message": "You cannot call 'EntityManager.findOne()' with empty 'where' parameter",
      "stack":
          Error: You cannot call 'EntityManager.findOne()' with empty 'where' parameter
              at EntityValidator.validateEmptyWhere (.../node_modules/@mikro-orm/core/entity/EntityValidator.js:70:19)
              at SqlEntityManager.findOne (../node_modules/@mikro-orm/core/EntityManager.js:223:24)
              at processTicksAndRejections (node:internal/process/task_queues:96:5)
              at async SqlEntityManager.findOneOrFail (../node_modules/@mikro-orm/core/EntityManager.js:254:24)
```

I know that this should work in different transactions, but I don't understand why the flush not working.

If I chain promises in the last service, it works.

**Versions**

| Dependency | Version |
| - | - |
| node | 16 |
| typescript | 4.3.5 |
| mikro-orm | 4.5.7 |
| postgresql | 4.5.7 |

**Comment by notmedia** [comment]
Related to https://github.com/mikro-orm/mikro-orm/issues/470

**Comment by B4nan** [comment]
I don't understand what you don't understand :] The issue you linked is exactly about this, and states this is a wontfix. Nothing changed, you simply cannot use the same EM (= same UoW instance) in parallel. 

https://github.com/mikro-orm/mikro-orm/issues/470#issuecomment-611238814

The problem is that the flush process mutates the identity map, so at one point in time, the same UoW instance tries to work with the data that should be already persisted (and therefore have the PK available), but in fact it is not there, so it fails.

You need to fork for each context/"thread".
